### PR TITLE
feat(arx) : new app

### PR DESCRIPTION
**Pull request by floriansipp**

### CODE REVIEW

**Comment by greut** [comment]
This is what I get.

```
/docker-entrypoint.d/2-run-app.sh: line 22: [: ==: unary operator expected
Running  as chorus on GPU...
OpenJDK 64-Bit Server VM warning: Option MaxRAMFraction was deprecated in version 10.0 and will likely be removed in a future release.
org.eclipse.swt.SWTError: No more handles [gtk_init_check() failed]
        at org.eclipse.swt.SWT.error(SWT.java:4749)
        at org.eclipse.swt.widgets.Display.createDisplay(Display.java:1130)
        at org.eclipse.swt.widgets.Display.create(Display.java:1070)
        at org.eclipse.swt.graphics.Device.<init>(Device.java:175)
        at org.eclipse.swt.widgets.Display.<init>(Display.java:636)
        at org.eclipse.swt.widgets.Display.<init>(Display.java:627)
        at org.deidentifier.arx.gui.Main.main(Unknown Source)
        at org.deidentifier.arx.gui.Main.main(Unknown Source)
Authorization required, but no authorization protocol specified

Exception in thread "main" java.awt.AWTError: Can't connect to X11 window server using 'workbench-sample.default:80' as the value of the DISPLAY variable.
        at java.desktop/sun.awt.X11GraphicsEnvironment.initDisplay(Native Method)
```

**Comment by floriansipp** [comment]
Your computer fails, mine is working, this is weird i like it

**Review by iDmple — APPROVED** [review]

**Review by iDmple — APPROVED** [review]

File: apps/arx/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+WORKDIR /apps/${APP_NAME}
+
+ARG DEBIAN_FRONTEND=noninteractive
+RUN --mount=type=cache,target=/var/cache/apt,sharing=locked \
+    --mount=type=cache,target=/var/lib/apt,sharing=locked \
+    apt-get update -qy && \
+    apt-get install --no-install-recommends -qy \
+    ca-certificates curl xdg-utils desktop-file-utils \
+    shared-mime-info libswt-gtk-4-java dbus-x11 at-spi2-core && \
+    curl -sSL "https://arx.deidentifier.org/?ddownload=2135" -o arxinstaller && \
```
Comment (greut) [inline apps/arx/Dockerfile:17]: Did you try installing the Runnable Jar files? https://arx.deidentifier.org/downloads/

It feels overkill to download a full blown installer meant for humans and run it as _unattended_ when there is, what appears to be, a simpler way.

Comment (floriansipp) [reply apps/arx/Dockerfile:17]: Yes but it had some dependencies that i didn't found at the time so i went to the easiest solution.

File: apps/arx/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ENV PROCESS_NAME="ARX-launcher"
+ENV APP_DATA_DIR_ARRAY=".local"
+ENV DATA_DIR_ARRAY=""
+
+# Install vgl + whatever is necessary for chorus use
+RUN --mount=type=bind,source=./core,target=/tmp/core_scripts,readonly=false \
+    pushd /tmp/core_scripts && \
+    pushd /tmp/core_scripts/utilities && \
+    ./chorus-utils.sh && \
+    popd && mv entrypoint/docker-entrypoint.sh / && \
```
[the remark this answers is not in the document]
Comment (floriansipp) [reply apps/arx/Dockerfile:40]: Sure, but in that case i need to modify the chorus-utils script again, atm it works with the assumption that the folder the script is launched from is the one where the script is.

[the remark this answers is not in the document]
Comment (greut) [reply apps/arx/Dockerfile:40]: that's a strong assumption. `dirname "$0"` should give you its current location.

[the remark this answers is not in the document]
Comment (floriansipp) [reply apps/arx/Dockerfile:40]: i like my assumptions like my coffee 
(joke aside, correcting that, will push a PR before validating this one and will apply the modification to the other images)

File: apps/arx/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    update-desktop-database ~/.local/share/applications/ && \ 
+    update-mime-database ~/.local/share/mime/ && \ 
+    gtk-update-icon-cache ~/.local/share/icons/hicolor/ -t && \ 
+    chmod 755 arxinstaller && \
+    ./arxinstaller --mode unattended && \ 
+    apt-get remove -y --purge curl && \
+    apt-get autoremove -y --purge 
+
+ENV APP_SPECIAL="no"
+ENV APP_CMD="/opt/ARX-3.9.1/ARX-launcher.run"
```
[the remark this answers is not in the document]
Comment (floriansipp) [reply apps/arx/Dockerfile:30]: Fair

File: apps/arx/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
+# syntax=docker/dockerfile:1
+FROM ubuntu:24.04
+SHELL ["/bin/bash", "-xe", "-o", "pipefail", "-c"]
+
+ARG APP_NAME
+ARG APP_VERSION
+
+WORKDIR /apps/${APP_NAME}
```
[the remark this answers is not in the document]
Comment (floriansipp) [reply apps/arx/Dockerfile]: Not really , will use /tmp for those images where it's just downloading something from now on

**Review by greut — APPROVED** [review]

**Pull request merged**